src/scripts/main.js

```diff
@@ -1,3 +1,110 @@
 'use strict';
 
-// write code here
+class Table {
+  TBODY = document.querySelector('.field tbody');
+  APPEND_ROW_BUTTON = document.querySelector('.append-row');
+  REMOVE_ROW_BUTTON = document.querySelector('.remove-row');
+  APPEND_COLUMN_BUTTON = document.querySelector('.append-column');
+  REMOVE_COLUMN_BUTTON = document.querySelector('.remove-column');
+
+  rowSize = 4;
+  cellsSize = 4;
+
+  constructor() {
+    this.getTableSize();
+    this.addListeners();
+  }
+
+  getTableSize() {
+    this.ROWS = this.TBODY.querySelectorAll('tr');
+
+    if (this.ROWS.length > 0) {
+      const cells = this.ROWS[0].cells;
+
+      this.rowSize = this.ROWS.length;
+      this.cellsSize = cells.length;
+    }
```

Comment:
The `getTableSize` method is called multiple times which results in unnecessary DOM querying. Consider storing the table size in a variable and updating it only when a row or column is added or removed.